api/src/main/java/com/strategyobject/substrateclient/api/pallet/parachain_system/ParachainSystem.java

```diff
@@ -0,0 +1,81 @@
+package com.strategyobject.substrateclient.api.pallet.parachain_system;
+
+import com.strategyobject.substrateclient.pallet.annotation.*;
+import com.strategyobject.substrateclient.rpc.api.primitives.Hash;
+import com.strategyobject.substrateclient.scale.ScaleType;
+import com.strategyobject.substrateclient.scale.annotation.Scale;
+import com.strategyobject.substrateclient.scale.annotation.ScaleReader;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.math.BigInteger;
+
+
+@Pallet("ParachainSystem")
+public interface ParachainSystem {
+    /**
+     * The validation function has been scheduled to apply.
+     */
+    @Event(index = 0)
+    @Getter
+    @Setter
+    @ScaleReader
+    class ValidationFunctionStored {}
+
+    /**
+     * The validation function was applied as of the contained relay chain block number.
+     */
+    @Event(index = 1)
+    @Getter
+    @Setter
+    @ScaleReader
+    class ValidationFunctionApplied {
+        @Scale(ScaleType.U32.class)
+        private Long relayChainBlockNum;
+    }
+
+    /**
+     * The relay-chain aborted the upgrade process.
+     */
+    @Event(index = 2)
+    @Getter
+    @Setter
+    @ScaleReader
+    class ValidationFunctionDiscarded {}
+
+    /**
+     * An upgrade has been authorized.
+     */
+    @Event(index = 3)
+    @Getter
+    @Setter
+    @ScaleReader
+    class UpgradeAuthorized {
+        private Hash codeHash; //H256
```

Comment:
We may need to use a ByteArray instead of a Hash for the Hash type, will update when I can confirm that's the case

Reply:
Left a comment about this on your PR @abadaa I think we're okay provided the type in question is one of `@AutoRegister(types = {Hash256.class, Hash.class, BlockHash.class, CallHash.class})`

Reply:
Got it, then I'll go ahead and resolve this since this looks like a Hash256